src/bucket/test/BucketTests.cpp

```diff
@@ -319,6 +328,71 @@ generateDifferentAccount(std::vector<LedgerEntry> const& others)
     }
 }
 
+TEST_CASE("merges proceed old-style despite newer shadows",
+          "[bucket][bucketmaxprotocol]")
+{
+    VirtualClock clock;
+    Config const& cfg = getTestConfig();
+    Application::pointer app = createTestApplication(clock, cfg);
+    auto& bm = app->getBucketManager();
+    auto vers = getAppLedgerVersion(app);
+    LedgerEntry liveEntry = generateAccount();
+    LedgerEntry otherLiveA = generateDifferentAccount({liveEntry});
+
+    // Proto 10 bucket
+    auto b10first = Bucket::fresh(bm, vers - 2, {}, {liveEntry}, {},
```

Comment:
also these `vers - 2` things seem to me .. not more legible than `10`, though I gather this was in response to an earlier complaint about literal `10` being written there. Consider this my disagreement with the other reviewer, at least in this one test (which is very specifically testing behaviours we expect for each of those versions).

Maybe split the difference and initialize some constants like `const v12 = FIRST_PROTOCOL_SHADOWS_REMOVED`, `const v11 = v12 - 1`, `const v10 = v11 - 1` and use the `v10` .. `v12` variables to indicate that they're versions? Bleh. I feel like such a nit-picker even mentioning this. But I'm having to remind me self on each stanza which version each expression denotes.

Reply:
fixed